OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -58,6 +60,27 @@ public void WorldLoaded(World w, WorldRenderer wr)
 
 			foreach (var cell in Map.AllCells)
 				UpdateCell(cell);
+
+			// Build the sprite layer dictionary for rendering resources
+			// All resources that have the same palette must also share a sheet and blend mode
```

Comment:
`ResourceLayer` already has the exact same piece of code. Seems like SmudgeLayer isn't too far off either.
Looks like they duplicate a bunch of other code as well. I can't suggest a place at the moment, but those duplicate code snippets should be extracted to a common location and called from there.